### PR TITLE
sort baseline output by file and resource

### LINKED ISSUE #3119: baseline output can change resource order for each run
If I generate a baseline file and I have then made some improvements to my Terraform code and I run the baseline again. What I am finding is that the order of the resources for each file can often change which then shows up as a diff against the prevous baseline file - when in reality nothing has change but the order of the resources in the findings array in the baseline output file 

I was wondering could the findings array just be sorted before being output? Then the resource order should be fixed and any actual diffs should be real changes to check_ids (which is sorted already) or new resources being added?

e.g. this is a diff from two runs of generating a baseline file nothing has actually change just resources moved around in the array.

```
@@ -100,13 +100,12 @@
             "file": "/main.tf",
             "findings": [
                 {
-                    "resource": "aws_s3_bucket.canary_artifacts",
+                    "resource": "aws_s3_bucket.backups",
                     "check_ids": [
                         "CKV2_AWS_6",
                         "CKV_AWS_144",
                         "CKV_AWS_145",
-                        "CKV_AWS_18",
-                        "CKV_AWS_21"
+                        "CKV_AWS_18"
                     ]
                 },
                 {
@@ -119,12 +118,13 @@
                     ]
                 },
                 {
-                    "resource": "aws_s3_bucket.lambdas",
+                    "resource": "aws_s3_bucket.canary_artifacts",
                     "check_ids": [
                         "CKV2_AWS_6",
                         "CKV_AWS_144",
                         "CKV_AWS_145",
-                        "CKV_AWS_18"
+                        "CKV_AWS_18",
+                        "CKV_AWS_21"
                     ]
                 },
                 {
@@ -137,7 +137,7 @@
                     ]
                 },
                 {
-                    "resource": "aws_s3_bucket.backups",
+                    "resource": "aws_s3_bucket.lambdas",
                     "check_ids": [
                         "CKV2_AWS_6",
                         "CKV_AWS_144",
```

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

The output is now sorted by file path and for each file by resource.

Fixes #3119

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/output/baseline.py [hunk 1]
```diff
@@ -56,9 +58,9 @@ def to_dict(self) -> dict[str, Any]:
             formatted_findings = []
             for finding in findings:
                 formatted_findings.append({"resource": finding["resource"], "check_ids": finding["check_ids"]})
-            failed_checks_list.append({"file": file, "findings": formatted_findings})
+            failed_checks_list.append({"file": file, "findings": sorted(formatted_findings, key=itemgetter("resource"))})
 
-        resp = {"failed_checks": failed_checks_list}
+        resp = {"failed_checks": sorted(failed_checks_list, key=itemgetter("file"))}
```
Comment (nimrodkor) [inline checkov/common/output/baseline.py:63]: @gruebel Within the same files, will the resources keep their order?

File: tests/common/output/test_baseline.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import argparse
```
Comment (itai1357) [inline tests/common/output/test_baseline.py:1]: as you want to affect also the order of the resources in some specific file, aren't better we that at least one of the files will continue one then 1 resource, while yor new logic will indeed affect the order?

**Review by itai1357 — APPROVED** [review]
Nice! approved with one comment regarding the test

**Review by schosterbarak — APPROVED** [review]

File: checkov/common/output/baseline.py (hunk 1 above)
Comment (gruebel) [reply checkov/common/output/baseline.py:63]: yes, I made sure of it, two lines up `sorted(formatted_findings, key=itemgetter("resource"))`

File: tests/common/output/test_baseline.py (hunk 2 above)
Comment (gruebel) [reply tests/common/output/test_baseline.py:1]: yeah good point 😄

**Pull request merged**